### PR TITLE
[spec] Rename simd->vec types

### PR DESCRIPTION
**Pull request by rossberg**
This does the analogous renaming from simd to vec in spec.

### CODE REVIEW

File: document/core/exec/numerics.rst [hunk 1]
```diff
@@ -180,13 +180,13 @@ When a number is stored into :ref:`memory <syntax-mem>`, it is converted into a
...
+.. index:: vector numbers, shape
 .. _aux-lanes:
 
-SIMD
-....
+Vector numbers
+..............
 
-SIMD values have the same underlying representation as an |i128|. They can also be interpreted as a sequence numeric values packed into a |V128| with a particular |shape|.
+Vector numbers have the same underlying representation as an |i128|. They can also be interpreted as a sequence numeric values packed into a |V128| with a particular |shape|.
```
Comment (ngzhian) [inline document/core/exec/numerics.rst:189]: Not part of your change but I noticed this typo: "sequence numeric values" => "sequence of numeric values"

File: document/core/syntax/values.rst [hunk 2]
```diff
@@ -145,14 +145,17 @@ Conventions
...
-   pair: abstract syntax; packed simd number
-.. _syntax-simd:
 
-Fixed-Width SIMD
-~~~~~~~~~~~~~~~~
+.. index:: ! vector numbers, integer, floating-point, lane, SIMD
+   pair: abstract syntax; vector
+.. _syntax-vecnum:
+
+Vector Numbers
```
Comment (ngzhian) [inline document/core/syntax/values.rst:153]: Vector Numbers sound a bit funny to me. Can we consider any other names? How about, Vector of numbers or Numeric vectors?

**Comment by ngzhian** [comment]
Adding Deepti as a reviewer too, as we are making more visible spec name changes.

File: document/core/exec/numerics.rst (hunk 1 above)
Comment (rossberg) [reply document/core/exec/numerics.rst]: Fixed.

File: document/core/syntax/values.rst (hunk 2 above)
Comment (rossberg) [reply document/core/syntax/values.rst]: Yes, that sounds better. Changed.

**Review by rossberg — COMMENTED** [review]
Sounds good. @dtig, this is an editorial change that essentially renames the notion of "SIMD types/values" to "vector types/values", to avoid the sort of category error that comes with describing data itself as "SIMD" and to be better aligned with the concrete type name "v128". The analogous change has already been applied to the interpreter (#513).

**Comment by rossberg** [comment]
@dtig, friendly ping.

**Review by dtig — APPROVED** [review]

**Comment by dtig** [comment]
Sorry for the delay, I missed the notifications on this one.

**Comment by rossberg** [comment]
Thanks @dtig! @ngzhian, are there open questions on your end?

**Comment by ngzhian** [comment]
No open questions, thanks for checking. LGTM!

**Review by ngzhian — APPROVED** [review]

**Pull request merged**